sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/README.md

```diff
@@ -8,112 +8,207 @@ The authentication events trigger for Azure Functions allows you to implement a
+```cs
+[FunctionName("onTokenIssuanceStart")]
+public static WebJobsAuthenticationEventResponse Run(
+    [WebJobsAuthenticationEventsTrigger] WebJobsTokenIssuanceStartRequest request, ILogger log)
+{
+    try
+    {
+        if (request.RequestStatus == WebJobsAuthenticationEventsRequestStatusType.Successful)
+        {
+            request.Response.Actions.Add(
+                new WebJobsProvideClaimsForToken(
+                    new WebJobsAuthenticationEventsTokenClaim("dateOfBirth", "01/01/2000"),
+                    new WebJobsAuthenticationEventsTokenClaim("customRoles", "Writer", "Editor"),
+                    new WebJobsAuthenticationEventsTokenClaim("apiVersion", "1.0.0"),
+                    new WebJobsAuthenticationEventsTokenClaim(
+                        "correlationId", 
+                        request.Data.AuthenticationContext.CorrelationId.ToString())));
+        }
+        else
+        {
+            log.LogInformation(request.StatusMessage);
+        }
+        return request.Completed();
+    }
+    catch (Exception ex) 
+    { 
+        return request.Failed(ex);
+    }
+}
+```
 
-If you want to test your API against some other client that is not Microsoft Entra authentication events service, like using Postman, you can configure an _optional_ application setting:
+### Build and run the project locally
 
-* **AuthenticationEvents__CustomCallerAppId** - the guid of your desired client. If not provided, `99045fe1-7639-4a75-9d4a-577b6ca3810f` is assumed.
+The project has been created, and the sample code has been added. Using your IDE, we need to build and run the project locally to extract the local function URL.
 
-#### Have the trigger validate the token
+---
 
-In local environments or environments that aren't hosted in the Azure Function service, the trigger can do the token validation. Set the following application settings in the [local.settings.json](https://learn.microsoft.com/azure/azure-functions/functions-develop-local#local-settings-file) file:
+### Run the function locally (recommended)
 
-* **AuthenticationEvents__TenantId** - your tenant ID
-* **AuthenticationEvents__AudienceAppId** - the same value as "Allowed audience" in option 1.
-* **AuthenticationEvents__CustomCallerAppId** (_optional_) - the guid of your desired client. If not provided, `99045fe1-7639-4a75-9d4a-577b6ca3810f` is assumed.
+It's a good idea to test the function locally before deploying it to Azure. We can use a dummy JSON body that imitates the request that Microsoft Entra ID sends to your REST API. Use your preferred API testing tool to call the function directly.
 
-An example `local.settings.json` file:
+1. In your IDE, open *local.settings.json* and replace the code with the following JSON. We can set `"AuthenticationEvents__BypassTokenValidation"` to `true` for local testing purposes.
 
 ```json
 {
```

Comment:
This can stay as-is.  We don't have a good way to validate non-C# code for these.  We'll just have to ensure that we keep it up to date over time.